platform.py

```
@@ -14,223 +14,449 @@
+        cmd = [
+            python_exe,
+            idf_tools_path,
+            "--quiet",
+            "--non-interactive",
+            "--tools-json",
+            tools_json_path,
+            "install"
+        ]
+
+        try:
+            with open(os.devnull, 'w') as devnull:
```

Ruff (0.11.9)

Local variable `devnull` is assigned to but never used

Remove assignment to unused variable `devnull`

(F841)